diff --git a/google_recaptcha.admin.inc b/google_recaptcha.admin.inc
index 46ad4b9..1e3d129 100644
--- a/google_recaptcha.admin.inc
+++ b/google_recaptcha.admin.inc
@@ -147,6 +147,8 @@ function google_recaptcha_forms($form, &$form_state) {
         }
     }
 
+    drupal_alter('google_recaptcha_available_forms', $available_forms);
+
     // create form
     $form['#prefix'] = '<p>' . t('Check forms which You want to protect with Google reCAPTCHA') . '</p>';
     $form['#prefix'] .= '<div class="description">' . t('For security usability reasons - forms for user registration, recovery pass and login always protected!') . '</div>';
diff --git a/google_recaptcha.api.php b/google_recaptcha.api.php
new file mode 100644
index 0000000..040469c
--- /dev/null
+++ b/google_recaptcha.api.php
@@ -0,0 +1,37 @@
+
+<?php
+/**
+ * @file
+ * Hooks provided by Google Recaptcha module.
+ */
+
+/**
+ * @addtogroup hooks
+ * @{
+ */
+
+/**
+ * Alters the list of always protected forms.
+ *
+ * @param array $always_protected
+ *   List of always protected forms.
+ */
+function hook_google_recaptcha_always_protect_alter(&$always_protected) {
+  // Add search form to the list of always protected forms.
+  $always_protected[] = 'search_form';
+}
+
+/**
+ * Alters the list of available forms.
+ *
+ * @param array $available_forms
+ *   List of available forms.
+ */
+function hook_google_recaptcha_available_forms_alter(&$available_forms) {
+  // Add search form to the list of always protected forms.
+  $always_protected['search_'] = t('Search');
+}
+
+/**
+ * @} End of "addtogroup hooks".
+ */
diff --git a/google_recaptcha.module b/google_recaptcha.module
index 6a9e7a6..fa4608d 100644
--- a/google_recaptcha.module
+++ b/google_recaptcha.module
@@ -98,6 +98,8 @@ function google_recaptcha_form_alter(&$form, &$form_state, $form_id) {
         'user_pass',
     );
 
+    drupal_alter('google_recaptcha_always_protect', $always_protected);
+
     if (in_array($form_id, $always_protected)) {
         google_recaptcha_add_captcha($form, $form_id);
     }
